src/commands/deploy/setPreDeployTaskForDotnet.ts

```diff
@@ -0,0 +1,73 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { SiteConfig } from 'azure-arm-website/lib/models';
+import * as path from 'path';
+import { Uri } from 'vscode';
+import { ext } from 'vscode-azureappservice/out/src/extensionVariables';
+import * as constants from '../../constants';
+import { findFilesByFileExtension, mapFilesToQuickPickItems } from '../../utils/workspace';
+import { getWorkspaceSetting, updateWorkspaceSetting } from '../../vsCodeConfig/settings';
+import * as tasks from '../../vsCodeConfig/tasks';
+import { IDeployWizardContext } from "../createWebApp/setAppWizardContextDefault";
+
+export async function setPreDeployTaskForDotnet(context: IDeployWizardContext, siteConfig: SiteConfig): Promise<void> {
+    // don't overwrite preDeploy task if it exists
+    if (!getWorkspaceSetting<boolean>('configurePreDeployTasks', context.workspace.uri.fsPath) || getWorkspaceSetting<boolean>(constants.configurationSettings.preDeployTask, context.workspace.uri.fsPath)) {
+        return;
+    }
+
+    // assume that the csProj is in the root
+    let csProjFsPath: string = context.workspace.uri.fsPath;
+
+    const csProjInRoot: Uri[] = await findFilesByFileExtension(context.workspace.uri.fsPath, 'csproj');
+    let csProjInSubfolders: Uri[] = [];
+
+    // if there was no .csproj in the root space, try to find it within subfolders
+    if (csProjInRoot.length === 0) {
+        // to have a recursive search of the opened workspace, pass in undefined rather than the fsPath
```

Comment:
This is completely recursive? That could really slow down deployment if you have a big workspace and I don't think we want to do that.